lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1151,6 +1142,23 @@ void TypeAnalysis::visitAtenLinearOp(AtenLinearOp op,
   incorporateKnowledge(op->getResult(0), knowledge);
 }
 
+template <typename OpTy> void TypeAnalysis::visitEmbeddingBagOp(OpTy op) {
```

Comment:
This doesn't need to be a template. The body of this function works if the input of the function is an `Operation *op`. Removing the `template` also allows you to simplify the code above. You should be able to do:

```c++
if (isa<Aten_EmbeddingBagOp, AtenEmbeddingBagPaddingIdxOp>(op)) {
  visit...;
return;
}
```

Reply:
makes sense. will make a change